### PR TITLE
[DBAAS-103] Single-Tenant security initial contribs

### PR DESCRIPTION
**Pull request by tchughesiv**
Signed-off-by: Tommy Hughes <tohughes@redhat.com>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: api/v1alpha1/dbaasconfig_types.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (bmozaffa) [inline api/v1alpha1/dbaasconfig_types.go:1]: We are hoping to have dbaasprovider.go contain all types that are of interest to database providers, and we are then referencing those from other files. I think this should go there too.

File: config/manifests/bases/dbaas-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -7,20 +7,25 @@ metadata:
...
   customresourcedefinitions:
     owned:
+    - description: DBaasConfig is the Schema for the dbaasconfigs API
+      displayName: DBaas Config
+      kind: DBaasConfig
+      name: dbaasconfigs.dbaas.redhat.com
+      version: v1alpha1
     - description: DBaaSConnection is the Schema for the dbaasconnections API
-      displayName: DBaaS Connection
+      displayName: DBaa SConnection
```
Comment (bmozaffa) [inline config/manifests/bases/dbaas-operator.clusterserviceversion.yaml:23]: Haven't figured out how to fix this, but have been manually correcting it after it's generated.

File: config/manifests/bases/dbaas-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -7,20 +7,25 @@ metadata:
...
       kind: DBaaSConnection
       name: dbaasconnections.dbaas.redhat.com
       version: v1alpha1
     - description: DBaaSInventory is the Schema for the dbaasinventory API
-      displayName: DBaaS Inventory
+      displayName: DBaa SInventory
```
Comment (bmozaffa) [inline config/manifests/bases/dbaas-operator.clusterserviceversion.yaml:28]: Haven't figured out how to fix this, but have been manually correcting it after it's generated.

**Review by bmozaffa — CHANGES_REQUESTED** [review]
Partial review!

File: config/manifests/bases/dbaas-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (tchughesiv) [reply config/manifests/bases/dbaas-operator.clusterserviceversion.yaml:23]: the fix is to switch kind from DBaaSConnection -> DBaa**s**Connection . 

For reference, see DBaasConfig above or [DBaasTenant PR](https://github.com/RHEcosystemAppEng/dbaas-operator/pull/20/files#diff-a12d4f08001ca5f2bdddf8fd2519349ad3549f9057c981a1660c4f36dc637bbcR28).

I can fix this in a separate PR.

Comment (bmozaffa) [reply config/manifests/bases/dbaas-operator.clusterserviceversion.yaml:23]: But that's not really a fix, it's DBaaS :-)

Comment (tchughesiv) [reply config/manifests/bases/dbaas-operator.clusterserviceversion.yaml:23]: will manually fix for now

**Pull request closed without merging**